### ISSUE TITLE
Reduced performance in Catboost v0.16.2 as compared to v0.15.2

### ISSUE THREAD

**Issue by groverpr** [catboost/catboost#945]
Problem: The performance of latest version of Catboost with mostly default hyperparameters is lower than the 0.15.2 version for binary classification problem with ROC-AUC as evaluation metric. 

I am using Catboost to create baseline scores for[ IEEE Fraud Detection Kaggle Competetion](https://www.kaggle.com/c/ieee-fraud-detection). Keeping everything else same, the 0.15.2 version gives auc score of 94.4 and 0.16.2 gives 92.4 on holdout set. It's a big difference. Below is the link to an example notebook where I have shown the performance differences on IEEE dataset. 

https://github.com/groverpr/Machine-Learning/blob/master/catboost/catboost_version_diff.ipynb

catboost version: 0.16.2
Operating System: Linux EC2
CPU: 1 used (total 32)
GPU: 1 used (total 8)

**Comment by khrisanfov** [comment]
I think it is similar problem to https://github.com/catboost/catboost/issues/925 that I opened. There is definitely something wrong in 16.2. I temporarily switched back to 15.2.

**Comment by AdrianAntico** [comment]
Pretty sure v0.15.2 is not longer working properly either. I saw the same issue with 0.16.2 on 8-9-2019 and now I'm seeing issues with v0.15.2 today (08-11-2019).

**Comment by AdrianAntico** [comment]
I'd like to add to my previous comment. I installed v16.1 and tested it out. It crashed my R session. Tried a few more times with different eval_metrics, all crashed. THEN I switched the task_type to CPU and it worked fine. I'm going to see if it has something to do with my GPU or the associated drivers.

**Comment by annaveronika** [comment]
@AdrianAntico Could you please create a new issue and provide the full code that you were running? And also please try the latest version 0.16.3

**Comment by annaveronika** [comment]
@groverpr Thanks for the report, we are looking into this!

**Comment by groverpr** [comment]
@annaveronika I tested the code on 0.16.3. It gives similar score as 0.16.2 and has the same issue. The results are added in the [same](https://github.com/groverpr/Machine-Learning/blob/master/catboost/catboost_version_diff.ipynb) notebook. 

I also notice that the training and validation AUC curves reach a higher score (0.945) during fitting. But when I seperately calculate validation score on the fitted model using sklearn roc_auc_score(), it reduces by 2% ( to 0.924). I am using eval_set for early stopping and use_best_model=True. I don't observe this difference in v.0.15.2 and in that version the score during training matches with final score on trained mode.

**Comment by annaveronika** [comment]
Please set eval_metric='AUC:use_weights=false', I think we set it to true by default in last few releases, which is not what you want. We'll return the previous behavior in the next minor release.

**Comment by khrisanfov** [comment]
@annaveronika it doesn not work, I checked it on flight delays dataset (#925)

**Comment by groverpr** [comment]
@annaveronika  Thanks. I did an experiment w/ and w/o `use_weights=false` in eval_metric AUC. There is no change in the results. 

I did an additional experiment on CPU vs. GPU and notice that keeping everything else same, CPU version gives 1-1.5% point lift in scores as compared to using GPU. 

Again experiment results are added to [this](https://github.com/groverpr/Machine-Learning/blob/master/catboost/catboost_version_diff.ipynb) notebook.

**Comment by annaveronika** [comment]
Thanks a lot for reporting this! There was a bug in python prediction when having combinations of categorical features. The bug appeared when you are predicting on more than 128 documents. It is fixed now in catboost 0.16.4, the release is out, please update from pypi.

**Comment by annaveronika** [comment]
Closing the issue, sorry for inconvenience!

**Comment by annaveronika** [comment]
And please use use_weights=False by default, currently all eval_metrics are weighted by default.
